### ISSUE TITLE
It would be very nice to have a SearchPatient by Patient Demographics

### ISSUE THREAD

**Issue by sehma-yannis** [1naturalway/brightree#14]
The package is awesome but having this functionality implemented would be excellent since sometimes getting the PatientID gets complicated

**Comment by nickcheek** [comment]
> The package is awesome but having this functionality implemented would be excellent since sometimes getting the PatientID gets complicated

I don't believe Brightree allows it in their API.  Date of Birth would be the closest they allow you to get to any demographics.

**Comment by sehma-yannis** [comment]
Brightree has it implemented in SOAP, you can do a SearchPatient() by FirstName, LastName and DOB among other parameters. I think you have that implemented in your own package.

**Comment by nickcheek** [comment]
> Brightree has it implemented in SOAP, you can do a SearchPatient() by FirstName, LastName and DOB among other parameters. I think you have that implemented in your own package.

Ah, I see, I thought by demographics you meant sex, race, etc...

**Comment by nickcheek** [comment]
> Brightree has it implemented in SOAP, you can do a SearchPatient() by FirstName, LastName and DOB among other parameters. I think you have that implemented in your own package.

I've created a pull request that will give you the option of having a custom request to the API so you can use any fields to search that you'd like by creating your own custom object to send to the api.

**Comment by sehma-yannis** [comment]
> > Brightree has it implemented in SOAP, you can do a SearchPatient() by FirstName, LastName, and DOB among other parameters. I think you have that implemented in your own package.
> 
> I've created a pull request that will give you the option of having a custom request to the API so you can use any fields to search that you'd like by creating your own custom object to send to the API.

Great! Thanks a lot! We'll test it tomorrow and let you know how it went! Thanks again!

**Comment by nickcheek** [comment]
It's not my package so you'll have to wait on one of the maintainers to merge the request before it'll be available to use.
